hierarchy/api.go

```diff
@@ -0,0 +1,138 @@
+package hierarchy
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+
+	"github.com/ONSdigital/dp-hierarchy-api/models"
+	"github.com/ONSdigital/go-ns/log"
+	"github.com/ONSdigital/go-ns/rchttp"
+)
+
+// API aggregates a client and URL and other common data for accessing the API
+type API struct {
```

Comment:
Consider looking to see if the hierarchy API client can be reused from go-ns

Reply:
So after looking at this for a while I've decided to not use the go-ns client library for a couple of reasons firstly it doesn't use the rchttp package therefore context is not been passed to client. Secondly urls are not parsed, while this code has both.

I'll create some tech debt stories to upgrade the client package for all our clients to use both rchttp package and parse the urls.

Reply:
👍